modules/nlu/src/views/lite/intentEditor/IntentEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import { AxiosInstance } from 'axios'
+import { NLU } from 'botpress/sdk'
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+
+import { NLUAPI } from '../../api'
+
+import Slots from './slots/Slots'
+import style from './style.scss'
+import { removeSlotFromUtterances, renameSlotInUtterances } from './utterances-state-utils'
+import { ContextSelector } from './ContextSelector'
+import IntentHint from './IntentHint'
+import { UtterancesEditor } from './UtterancesEditor'
+
+interface Props {
+  intent: string
+  api: NLUAPI
+  contentLang: string
+  showSlotPanel: boolean
+  axios: AxiosInstance
+}
+
+export const IntentEditor: FC<Props> = props => {
+  const [intent, setIntent] = useState<NLU.IntentDefinition>()
+
+  useEffect(() => {
+    props.api.fetchIntent(props.intent).then(setIntent)
+  }, [props.intent])
+
+  if (!intent) {
+    // TODO display a fetching state instead
+    return null
+  }
+
+  const saveIntent = (newIntent: NLU.IntentDefinition) => {
+    setIntent(newIntent)
+    props.api.createIntent(newIntent)
+  }
+
+  const handleUtterancesChange = (newUtterances: string[]) => {
+    const newIntent = { ...intent, utterances: { ...intent.utterances, [props.contentLang]: newUtterances } }
+    saveIntent(newIntent)
+  }
+
+  const handleSlotsChange = (slots: NLU.SlotDefinition[], { operation, name, oldName }) => {
+    let newUtterances = [...intent.utterances[props.contentLang]]
+    if (operation === 'deleted') {
+      newUtterances = removeSlotFromUtterances(newUtterances, name)
+    } else if (operation === 'modified') {
+      newUtterances = renameSlotInUtterances(newUtterances, oldName, name)
+    }
+
+    const newIntent = { ...intent, utterances: { ...intent.utterances, [props.contentLang]: newUtterances }, slots }
+    saveIntent(newIntent)
+  }
+
+  const utterances = (intent && intent.utterances[props.contentLang]) || []
+
+  return (
+    intent && (
```

Comment:
There is already a null check at the beginning which returns null, so this one is redundant